Lecture3/Task1/Task1L3.java

```diff
@@ -0,0 +1,246 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Task1L3 {
+
+    private static List<Voucher> listOfVouchers = new ArrayList<>();
+    private static List<Voucher> listOfSelection = new ArrayList<>();
+    private static Map<Integer, Voucher> mapOfSelection = new TreeMap<>();
+
+    public static void main(String[] args) {
+        createRandomVouchers();
+        int selection =  getUserSelection();
+        provideFirstOffer(listOfVouchers, selection);
+        String advanced = needAdvancedSearch();
+        if (advanced.equalsIgnoreCase("n")) {
+            selectVoucher();
+        }
+        if (advanced.equalsIgnoreCase("y")) {
+            searchDaysCount();
+            searchFood();
+            searchTransfer();
+            provideFinalOffer();
+        }
+    }
+
+    private static void createRandomVouchers() {
+        for (int i = 50; i != 0; i--) {
+            int creationType = (int)(Math.random()*3);
+            if (creationType == 0) {
```

Comment:
if you need to check constant values, use switch instead of if

Reply:
fixed